Maple2.Server.Game/PacketHandlers/ItemUseHandler.cs

```diff
@@ -38,4 +46,46 @@ public class ItemUseHandler : PacketHandler<GameSession> {
         
         session.Send(StoryBookPacket.Load(storyBookId));
     }
+
+    private static void HandleChatSticker(GameSession session, Item item) {
+        Dictionary<string, string> parameters = XmlParseUtil.GetParameters(item.Metadata?.Function?.Parameters);
+
+        if (!parameters.ContainsKey("id")) {
+            session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed));
+            return;
+        }
+
+        int stickerSetId = int.Parse(parameters["id"]);
+
+        int duration = 0;
+        if (parameters.TryGetValue(parameters["id"], out string? durationString)) {
+            int.TryParse(durationString, out duration);
+        }
+        
+        if (session.Player.Value.Unlock.StickerSets.ContainsKey(stickerSetId)) {
+            if (session.Player.Value.Unlock.StickerSets[stickerSetId] == long.MaxValue) {
+                session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed_already_exist));
+                return;
+            }
+
+            // does not expire
+            if (duration == 0) {
+                session.Player.Value.Unlock.StickerSets[stickerSetId] = long.MaxValue;
+            } else {
+                session.Player.Value.Unlock.StickerSets[stickerSetId] = Math.Min(session.Player.Value.Unlock.StickerSets[stickerSetId] + duration, long.MaxValue);
```

Comment:
Based on the error message ("- pack has shorter duration than what you have") I don't believe this is the right handling of it. We should be comparing:
1. `session.Player.Value.Unlock.StickerSets[stickerSetId]`
2. `Now + duration`

```cs
long existingTime = 0;
if (session.Player.Value.Unlock.StickerSets.ContainsKey(stickerSetId)) {
    existingTime = session.Player.Value.Unlock.StickerSets[stickerSetId];
    if (existingTime == long.MaxValue) {
        session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed_already_exist));
        return;
    }
}

long newTime = duration > 0 : DateTimeOffset.UtcNow.ToUnixTimeSeconds() + duration ? long.MaxValue;
if (newTime <= existingTime) {
    session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed_already_exist));
    return;
}

session.Player.Value.Unlock.StickerSets[stickerSetId] = newTime;
```

Reply:
I feel as though that error is misleading, particularly the 1 of the 3 reasons you pointed out.
`- pack has shorter duration than what you have`
This can't ever really happen. If have an expiring sticker set and you try to use another sticker set that also expires, you end up _adding_ onto the current time.
(i.e. Stickers expire at 12/25. You use a sticker set item that has a duration of 30 days. Your stickers now expire at 1/25)

Reply:
I see, maybe this then?
```cs
long existingTime = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
if (session.Player.Value.Unlock.StickerSets.ContainsKey(stickerSetId)) {
    existingTime = session.Player.Value.Unlock.StickerSets[stickerSetId];
    if (existingTime == long.MaxValue) {
        session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed_already_exist));
        return;
    }
}

long newTime = duration > 0 : existingTime + duration ? long.MaxValue;
if (newTime <= existingTime) {
    session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed_already_exist));
    return;
}

session.Player.Value.Unlock.StickerSets[stickerSetId] = newTime;
```